### PR TITLE
[FSTORE-939] Change name of Stream Feature Group's DeltaStreamer Job

### PR DESCRIPTION
**Pull request by davitbzh**
This PR adds/fixes/changes...
- please summarize your changes to the code 
- and make sure to include all changes to user-facing APIs

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [ ] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/engine/python.py [hunk 1]
```diff
@@ -963,36 +963,40 @@ def acked(err, msg):
...
         if (
             not isinstance(feature_group, ExternalFeatureGroup)
             and reset_offsets
             and not feature_group._multi_part_insert
         ):
-            if offline_write_options is not None and not offline_write_options.get(
-                "start_offline_backfill", True
+            if offline_write_options is not None and not (
+                offline_write_options.get("start_offline_materialization", True)
+                or offline_write_options.get("start_offline_backfill", True)
```
Comment (SirOibaf) [inline python/hsfs/engine/python.py:975]: This if statement doesn't work. Users will only set one of the two (Let's say `start_offline_materialization` to `False`). The other one is going to be empty and so the results is going to be True.

I think the easiest is to remove the default value and check if the keys exists.

File: python/hsfs/engine/python.py [hunk 2]
```diff
@@ -963,36 +963,40 @@ def acked(err, msg):
...
             ):
                 warnings.warn(
-                    "This is the first ingestion after an upgrade or backup/restore, running backfill job even though `start_offline_backfill` was set to `False`.",
+                    "This is the first ingestion after an upgrade or backup/restore, running materialization job even though `start_offline_materialization` was set to `False`.",
                     util.FeatureGroupWarning,
                 )
-            feature_group.backfill_job.run(
-                args=feature_group.backfill_job.config.get("defaultArgs", "")
+            feature_group.materialization_job.run(
```
Comment (SirOibaf) [inline python/hsfs/engine/python.py:981]: Isn't this misaligned? Looking at the code (assuming the if statement above is fixed), if the user doesn't want to start the job, we only print a warning, but then we keep going and we actually run the job nevertheless?

Or am I missign somthing?

File: python/hsfs/engine/python.py [hunk 3]
```diff
@@ -963,36 +963,40 @@ def acked(err, msg):
...
                 await_termination=offline_write_options.get("wait_for_job", False),
             )
         elif (
             not isinstance(feature_group, ExternalFeatureGroup)
             and offline_write_options is not None
-            and offline_write_options.get("start_offline_backfill", True)
+            and (
+                offline_write_options.get("start_offline_materialization", True)
+                or offline_write_options.get("start_offline_backfill", True)
+            )
```
Comment (SirOibaf) [inline python/hsfs/engine/python.py:992]: Same comemnt as above.

**Review by SirOibaf — CHANGES_REQUESTED** [review]
Generally speaking I would mention in the documentation that the `start_offline_backfill` is deprecated but still works.

File: python/hsfs/engine/python.py (hunk 2 above)
Comment (davitbzh) [reply python/hsfs/engine/python.py:981]: hmm, checking master and indentation is the same there. But the warning saying `running materialization job even though `start_offline_materialization` was set to `False`` I guess that was intended?

File: python/hsfs/feature_group.py [hunk 4]
```diff
@@ -1663,8 +1663,14 @@ def save(
                   whether or not to the save call should return only
                   after the Hopsworks Job has finished. By default it does not wait.
                 * key `start_offline_backfill` and value `True` or `False` to configure
-                  whether or not to start the backfill job to write data to the offline
-                  storage. By default the backfill job gets started immediately.
+                  whether or not to start the materialization job to write data to the offline
+                  storage. `start_offline_backfill` is deprecated. Use `start_offline_materialization` instead.
+                * key `start_offline_backfill` and value `True` or `False` to configure
+                  whether or not to start the materialization job to write data to the offline
+                  storage. `start_offline_backfill` is deprecated. Use `start_offline_materialization` instead.
```
Comment (SirOibaf) [inline python/hsfs/feature_group.py:1670]: Duplicated documentation

File: python/hsfs/engine/python.py (hunk 2 above)
Comment (SirOibaf) [reply python/hsfs/engine/python.py:978]: Ok I see what you mean but then the indentation is correct.

File: python/hsfs/engine/python.py [hunk 5]
```diff
@@ -1305,3 +1301,15 @@ def is_connector_type_supported(connector_type):
...
+    def _start_offline_materialization(offline_write_options):
+        if offline_write_options is not None:
+            if "start_offline_materialization" in offline_write_options:
+                return offline_write_options.get("start_offline_materialization")
+            elif "start_offline_backfill" in offline_write_options:
+                return offline_write_options.get("start_offline_backfill")
+            else:
+                return True
+        else:
+            return False
```
Comment (SirOibaf) [inline python/hsfs/engine/python.py:1315]: Shouldn't this be returning True? if there are no options, the default behavior is to start the materialization job.

**Review by SirOibaf — CHANGES_REQUESTED** [review]

**Review by SirOibaf — APPROVED** [review]

**Pull request merged**